### PR TITLE
various fixes in the code for the msvc build to work

### PR DESCRIPTION
**Pull request by LefterisJP**

### CODE REVIEW

File: test/c/test.cpp [hunk 1]
```diff
@@ -236,6 +240,7 @@ BOOST_AUTO_TEST_CASE(ethash_check_difficulty_check) {
 
 BOOST_AUTO_TEST_CASE(test_ethash_dir_creation) {
     ethash_blockhash_t seedhash;
+	memset(&seedhash, 0, 32);
```
Comment (debris) [inline test/c/test.cpp]: 242 - spaces
234 - tab

Comment (LefterisJP) [inline test/c/test.cpp]: hehe thanks for spotting them. Will fix them from emacs .. I have no idea what Visual studio does!

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,4 +18,4 @@ if (OpenCL_FOUND)
 	add_subdirectory(src/libethash-cl)
 endif()
 add_subdirectory(src/benchmark EXCLUDE_FROM_ALL)
-add_subdirectory(test/c EXCLUDE_FROM_ALL)
+add_subdirectory(test/c)
```
Comment (debris) [inline CMakeLists.txt]: why have you changed this? Just want to know.

File: test/c/CMakeLists.txt [hunk 3]
```diff
@@ -8,8 +31,9 @@ IF( Boost_FOUND )
 
     link_directories ( ${Boost_LIBRARY_DIRS} )
     file(GLOB HEADERS "*.h")
-    ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
-
+	if (NOT MSVC)
+		ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
+	endif()
```
Comment (debris) [inline test/c/CMakeLists.txt]: spaces and tabs in this file

File: test/c/CMakeLists.txt [hunk 4]
```diff
@@ -1,4 +1,27 @@
+if (NOT MSVC)
+	set (CMAKE_PREFIX_PATH "$ENV{BOOST_ROOT}")
```
Comment (debris) [inline test/c/CMakeLists.txt]: I'm not sure if we should force people to set BOOST_ROOT as they env variable. If we can remove this lines, we can specify BOOST_ROOT from command line:

```bash
cmake -DCMAKE_PREFIX_PATH=$BOOST_ROOT .
```

File: CMakeLists.txt (hunk 2 above)
Comment (LefterisJP) [inline CMakeLists.txt]: So .. the only way to get the tests to find, compile and link to the ETHASH lib is to run cmake as specified in https://github.com/ethereum/ethash/blob/master/test/c/test.sh

If you do that in Windows the generated Visual studio project does not contain the Test project at all. Removing this allows for the Test project to show up in Visual Studio. 

Perhaps there is a better way with CMake? Or perhaps this whole thing is pretty wrong for anything other than Ubuntu?

File: test/c/CMakeLists.txt (hunk 4 above)
Comment (LefterisJP) [inline test/c/CMakeLists.txt]: yeah this is probably a better idea even though most people in windows from what I read seem to set this environment variable.

But to be honest .. the best thing would be to use the same thing you made for the ETHCmakeDependecies in cpp-ethereum and have it download boost and add its root as the `CMAKE_PREFIX_PATH`.

We need to make it as automatic as possible in windows

**Pull request merged**